org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterAtmosphericPressure.java

```diff
@@ -52,40 +52,56 @@
     private Integer enhancedScale = null;
 
     @Override
-    public synchronized boolean initializeConverter() {
-        cluster = (ZclPressureMeasurementCluster) endpoint.getInputCluster(ZclPressureMeasurementCluster.CLUSTER_ID);
-        if (cluster == null) {
+    public boolean initializeDevice() {
+        ZclPressureMeasurementCluster serverCluster = (ZclPressureMeasurementCluster) endpoint
+                .getInputCluster(ZclPressureMeasurementCluster.CLUSTER_ID);
+        if (serverCluster == null) {
             logger.error("{}: Error opening device pressure measurement cluster", endpoint.getIeeeAddress());
             return false;
         }
 
         // Check if the enhanced attributes are supported
-        if (cluster.getScaledValue(Long.MAX_VALUE) != null) {
-            enhancedScale = cluster.getScale(Long.MAX_VALUE);
+        if (serverCluster.getScaledValue(Long.MAX_VALUE) != null) {
+            enhancedScale = serverCluster.getScale(Long.MAX_VALUE);
             if (enhancedScale != null) {
                 enhancedScale *= -1;
             }
         }
 
-        bind(cluster);
-
-        // Add a listener, then request the status
-        cluster.addAttributeListener(this);
-
-        // Configure reporting - no faster than once per second - no slower than 2 hours.
         try {
-            CommandResult reportingResponse;
-            if (enhancedScale != null) {
-                reportingResponse = cluster.setScaledValueReporting(1, REPORTING_PERIOD_DEFAULT_MAX, 0.1).get();
-                handleReportingResponse(reportingResponse, POLLING_PERIOD_DEFAULT, REPORTING_PERIOD_DEFAULT_MAX);
+            CommandResult bindResponse = bind(serverCluster).get();
+            if (bindResponse.isSuccess()) {
+                // Configure reporting - no faster than once per second - no slower than 2 hours.
+                CommandResult reportingResponse;
+                if (enhancedScale != null) {
+                    reportingResponse = serverCluster.setScaledValueReporting(1, REPORTING_PERIOD_DEFAULT_MAX, 0.1)
+                            .get();
+                    handleReportingResponse(reportingResponse, POLLING_PERIOD_DEFAULT, REPORTING_PERIOD_DEFAULT_MAX);
+                } else {
+                    reportingResponse = serverCluster.setMeasuredValueReporting(1, REPORTING_PERIOD_DEFAULT_MAX, 0.1)
+                            .get();
+                    handleReportingResponse(reportingResponse, POLLING_PERIOD_DEFAULT, REPORTING_PERIOD_DEFAULT_MAX);
+                }
             } else {
-                reportingResponse = cluster.setMeasuredValueReporting(1, REPORTING_PERIOD_DEFAULT_MAX, 0.1).get();
-                handleReportingResponse(reportingResponse, POLLING_PERIOD_DEFAULT, REPORTING_PERIOD_DEFAULT_MAX);
+                logger.error("{}: Error 0x{} setting server binding", endpoint.getIeeeAddress(),
+                        Integer.toHexString(bindResponse.getStatusCode()));
```

Comment:
I think we actually should return false if there's an error here. I suspect you've copied this from the converter I wrote originally which also returns true in the event of an exception, but it seems incorrect to me. WDYT?

Reply:
>  I suspect you've copied this from the converter I wrote originally
That is correct, I used the `ZigBeeConverterSwitchOnoff` as a reference. I can see now that it is also described in the `ZigBeeBaseChannelConverter.initializeDevice()` Javadoc that:
- this methos should return true if the device was configured correctly. Therefore, it looks that we should return `false` here. 
- this methos should set the `pollingPeriod` variabl to `POLLING_PERIOD_HIGH` if it fails. 

I will implement both changes.